plugins/Vestige/Vestige.cpp

```diff
@@ -820,9 +820,12 @@ void VestigeInstrumentView::noteOffAll( void )
 	m_vi->m_pluginMutex.lock();
 	if( m_vi->m_plugin != nullptr )
 	{
-		for( int key = 0; key <= MidiMaxKey; ++key )
+		for (int channel = 0; channel < 16; ++channel)
 		{
-			m_vi->m_plugin->processMidiEvent( MidiEvent( MidiNoteOff, 0, key, 0 ), 0 );
+			for(int key = 0; key <= MidiMaxKey; ++key)
```

Comment:
```suggestion
			for (int key = 0; key <= MidiMaxKey; ++key)
```
I wonder if the `<=` should be `<`?

Also, this method seems fairly inefficient. I wonder if there's a single MIDI message that turns all notes off? Though if so, the plugins would need to support it.

Reply:
For some reason, `MidiMaxKey` is 127, since it's supposed to be the index of the last key. Maybe I should rename MidiMaxKey to MidiNumKeys, set it to 128, and update all the code which uses it.

I believe there is a controller message which silences all notes. Maybe I can test if that works. But as you mentioned, I don't know how many plugins would support it, so having the emergency all notes off button send NoteOff signals to all keys could be a safer option.

Reply:
I just tested the All Notes Off signal (Control Change #123) on the two VSTs I have. Vital responds correctly, but OctaSine does not.

That's only two VSTs, so I don't know how common that is. But if it's not uncommon for VSTs to not support the All Notes Off signal, we might be forced to send NoteOff signals on all keys on all channels(?) Unless there is a way to determine if the VST supports it, but I'm not aware of any way.